workbench-core/accounts/src/services/accountService.test.ts

```diff
@@ -272,59 +247,119 @@ describe('AccountService', () => {
       subnetId: '',
       vpcId: '',
       awsAccountId: '123456789012',
-      id: 'sampleAccId'
+      id: 'sampleAccId',
+      updatedAt: '',
+      createdAt: ''
     };
 
     const accounts = [marshall(account)];
 
     const expectedList = [account];
 
-    const mockDDB = mockClient(DynamoDBClient);
     mockDDB.on(QueryCommand).resolves({
       Items: accounts
     });
 
     // OPERATE
-    const response = await accountService.getAccounts();
+    const response = await accountService.getAccountsForAccountHandler({
+      index: 'getResourceByCreatedAt',
+      key: { name: 'resourceType', value: 'account' }
+    });
 
     // CHECK
     expect(response).toEqual(expectedList);
   });
 
-  describe('getAccount', () => {
-    let dynamoMock: AwsStub<ServiceInputTypes, ServiceOutputTypes>;
-    let accountService: AccountService;
+  describe('getAccounts', () => {
+    let accountJson: Record<string, JSONValue>;
+    let paginationToken: string | undefined;
 
     beforeEach(() => {
-      dynamoMock = mockClient(DynamoDBClient);
-      accountService = new AccountService(process.env.STACK_NAME!);
+      accountJson = {
+        name: '',
+        cidr: '',
+        encryptionKeyArn: '',
+        envMgmtRoleArn: '',
+        environmentInstanceFiles: '',
+        externalId: '',
+        hostingAccountHandlerRoleArn: '',
+        stackName: '',
+        status: 'CURRENT',
+        subnetId: '',
+        vpcId: '',
+        awsAccountId: '123456789012',
+        id: 'sampleAccId',
+        updatedAt: '',
+        createdAt: ''
+      };
+
+      paginationToken = 'paginationToken';
+
+      jest.spyOn(DynamoDBService.prototype, 'getPaginatedItems').mockImplementation(() => {
+        return Promise.resolve({
+          data: [accountJson],
+          paginationToken
+        });
+      });
+    });
+
+    test('returns a paginated response of accounts', async () => {
+      const actualResponse = await accountService.getAccounts({});
+      expect(actualResponse).toEqual({
+        data: [AccountParser.parse(accountJson)],
+        paginationToken
+      });
+    });
+
+    describe('when there is no pagination token from dynamo', () => {
+      beforeEach(() => {
+        paginationToken = undefined;
```

Comment:
We don't need a`beforeEach` for this test right? There's only one test so `paginationToken` can just go on line 320.

Reply:
The `beforeEach` function is a place to set up the described scenario for the code. In this case, we're setting up a scenario where there is no pagination token. Using a before each means that we don't have to copy and past that set up for any other tests in this scenario since they would all need `pagionationToken` to equal `undefined`.

Reply:
Ah got it. Thanks for the explanation.